[safe-merge-queue] Should skip Windows/Linux checks in security branches

briannafan opened the pull request:
#### 961c9da7a4b6001b35e593c6f07c899d8e293476
<pre>
[safe-merge-queue] Should skip Windows/Linux checks in security branches
<a href="https://bugs.webkit.org/show_bug.cgi?id=264529">https://bugs.webkit.org/show_bug.cgi?id=264529</a>
<a href="https://rdar.apple.com/118203999">rdar://118203999</a>

Reviewed by Jonathan Bedard.

Only check when a commit is from WebKit project.
A temporary fix until we figure out why EWS statuses for those skipped tests are not updated in ews.webkit.org.

* Tools/CISupport/ews-build/steps.py:
(CheckStatusOfPR):
(CheckStatusOfPR.getQueueStatusFromList):

Canonical link: <a href="https://commits.webkit.org/270663@main">https://commits.webkit.org/270663@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bf9cf9c6abf9ade8fd98b63382f7857bbb467d0d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26007 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4619 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27290 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28107 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23822 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26325 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6382 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2048 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23890 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26261 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3499 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22411 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28687 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3119 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23363 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29422 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23735 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23740 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27305 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3150 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1361 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/25775 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4541 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6268 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3609 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3470 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -2534,7 +2533,12 @@ def getQueueStatusFromList(self, change_id, pr_number):
             yield self._addToLog('stdio', f'Accessed {url} with unexpected status code {response.status_code}.\n')
             return defer.returnValue(False if response.status_code // 100 == 4 else None)
 
-        for queue in self.QUEUES_FOR_SAFE_MERGE_QUEUE:
```
You should add a `#FIXME` comment here, because I think it's both of our assessment that you shouldn't need this, and skipping a queue should be a valid state for Safe Merge Queue to continue.

briannafan committed:
[safe-merge-queue] Should skip Windows/Linux checks in security branches
https://bugs.webkit.org/show_bug.cgi?id=264529
rdar://118203999

Reviewed by Jonathan Bedard.

Only check when a commit is from WebKit project.
A temporary fix until we figure out why EWS statuses for those skipped tests are not updated in ews.webkit.org.

* Tools/CISupport/ews-build/steps.py:
(CheckStatusOfPR):
(CheckStatusOfPR.getQueueStatusFromList):

Canonical link: https://commits.webkit.org/270663@main
